src/main/kotlin/ch/hsr/dsl/dwrtc/util/TomP2P.kt

```diff
@@ -19,3 +20,27 @@ fun BaseFuture.onSuccess(emitter: () -> Unit) {
 fun BaseFuture.onFailure(emitter: (failedReason: String) -> Unit) {
     this.onComplete { result -> if (result.isFailed) emitter(result.failedReason()) }
 }
+
+fun <T, U> FutureGet.onGetCustom(emitter: (data: U?, future: Future) -> Unit, transformer: (T) -> U) {
+    this.addListener(object : BaseFutureAdapter<FutureGet>() {
+        override fun operationComplete(future: FutureGet) {
+            val data = transformer(future.data()?.`object`() as T) ?: null
```

Comment:
IntelliJ gives me a notice on compilation:
Warning:(27, 68) Kotlin: Right operand of elvis operator (?:) is useless if it is null